src/chunk.h

```diff
@@ -118,6 +118,26 @@ typedef struct ChunkScanEntry
 	ChunkStub *stub;
 } ChunkScanEntry;
 
+/* To track a chunk move or copy activity */
+typedef struct ChunkMoveData
+{
+	/* numeric id for this chunk copy/move activity */
+	int32 id;
+	/* chunk to copy */
+	Chunk *chunk;
+	/* shared name used to name replication slot, publication and
+	 * subscription */
+	NameData operation_id;
+	/* from node */
+	const char *src_node;
```

Comment:
Maybe make these `NameData` members to make the struct self-contained instead of having references? This avoids potential lifetime issues.